### ISSUE TITLE
Findone() cannot work well using mongoDB.

### ISSUE THREAD

**Issue by xieyang784** [strongloop/loopback-connector-mongodb#374]
Env: windows 10, mongoDB(single instance, no replic)
findOne can work sometimes with right results, but with some 'where' situation return null.
```
          app.models.employee.findOne(
              // { where: { and: [{ userId: `${usr.id}` }, { phone: `${usr.phone}` }, { identity: 'employee' }, { state: true }] } },   // can't work, return null
             // { where: { and: [{ id: `${id}` }] } },   // work, return correct results
            { userId: `${usr.id}`, phone: `${usr.phone}`, identity: 'employee', state: true }, // work, return correct results
            (error, employeeInfo) => {
```
but, in the same file, I have another findOne function, that can work, here is the code:

```
              app.models.user.findOne(
                { where: { and: [{ orgName: `${employeeInfo.orgName}` }, { identity: 'company' }] } },  // work, return correct results
                (e, hirerInfo) => {
```

It's so wired, right? Thanks for your reply sooner.

**Comment by xieyang784** [comment]
Get it, it's about objectId.

**Comment by dhmlau** [comment]
@xieyang784 , just wanna clarify.  Your problem is solved or you figured out the problem is caused by the `usr.id` being a objectId type?

**Comment by xieyang784** [comment]
@dhmlau , I figured out and solved by using "loopback-objectid-mixin": "^1.0.1". Thanks for your reply.
